app/src/main/java/com/sopt/dive/signup/SignUpActivity.kt

```diff
@@ -0,0 +1,375 @@
+import androidx.compose.material3.Button
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.sopt.dive.R
+import com.sopt.dive.login.Greeting
+import com.sopt.dive.ui.theme.DiveTheme
+
+class SignUpActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        enableEdgeToEdge()
+        setContent {
+            DiveTheme {
+                Scaffold(modifier = Modifier.fillMaxSize()) {innerPadding ->
+                    SingUP(
+                        modifier = Modifier
+                            .padding(innerPadding)
+                            .fillMaxSize()
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Preview(showBackground = true)
+@Composable
+fun Preview() {
+    SingUP(Modifier.fillMaxSize())
+}
+
+@Composable
+fun SingUP(modifier: Modifier = Modifier) {
+    val context = LocalContext.current
+    val activity = context as? ComponentActivity
+
+    // 입력값 상태 관리
+    var idText by remember { mutableStateOf("") }
+    var pwText by remember { mutableStateOf("") }
+    var nicknameText by remember { mutableStateOf("") }
+    var drankText by remember { mutableStateOf("") }
+
+    // 에러 메시지 상태 관리
+    var idError by remember { mutableStateOf("") }
+    var pwError by remember { mutableStateOf("") }
+    var nicknameError by remember { mutableStateOf("") }
+    var drankError by remember { mutableStateOf("") }
+
+//    // Snackbar
+//    val scope = rememberCoroutineScope()
+//    val snackbarHostState = remember { SnackbarHostState() }
+
+
+    fun validateId(id: String): Boolean {
+        return when {
+            id.isEmpty() -> {
+                idError = "ID를 입력해주세요."
+                false
+            }
+            id.contains(" ") -> {
+                idError = "ID에는 공백을 사용할 수 없습니다."
+                false
+            }
+            id.length < 6 -> {
+                idError = "ID는 6글자 이상이어야 합니다."
+                false
+            }
+            id.length > 10 -> {
+                idError = "ID는 10글자 이하여야 합니다."
+                false
+            }
```

Comment:
검증 조건의 차이긴 한데.. 에러 메시지 구분을 다르게 가져가야 한다면 범위로 비교하는 방법도 있을 것 같아요!
```suggestion
            id.length !in 6..10 -> {
                idError = "ID는 6~10글자이어야 합니다."
                false
            }
```